js/src/utils.ts

```diff
@@ -55,21 +57,46 @@ export const calculateStakeAccountAddress = async (
 };
 
 /**
- * Get the stake account that has the most amount of SOL staked(heaviest)
+ * Get the stake account that has the most amount of SOL staked (heaviest)
  * @param snapshot Snapshot of the Solido stats
  * @returns Heaviest stake account
  */
 export const getHeaviestValidatorStakeAccount = (
   snapshot: Snapshot
 ): Snapshot['validatorsStakeAccounts'][0] => {
-  const sortedValidatorStakeAccounts = snapshot.validatorsStakeAccounts.sort(
-    (stakeAccountA, stakeAccountB) =>
-      stakeAccountB.balance.lamports
-        .sub(stakeAccountA.balance.lamports)
-        .toNumber()
-  );
+  let heaviestValidatorStakeAccount = snapshot.validatorsStakeAccounts[0];
 
-  const heaviestValidatorStakeAccount = sortedValidatorStakeAccounts[0];
+  snapshot.validatorsStakeAccounts.forEach((validatorStakeAccount) => {
+    if (
+      validatorStakeAccount.balance.lamports.gt(
+        heaviestValidatorStakeAccount.balance.lamports
+      )
+    ) {
+      heaviestValidatorStakeAccount = validatorStakeAccount;
+    }
+  });
 
   return heaviestValidatorStakeAccount;
 };
+
+/**
+ * Exchange SOL to stSOL
+ * @param snapshot Snapshot of the Solido stats
+ * @param amount SOL to exchange
+ */
+export const exchangeSol = (snapshot: Snapshot, amount: Lamports) => {
+  const exchangeRate = getExchangeRate(snapshot);
+
+  return new StLamports(amount.lamports.toNumber() / exchangeRate);
+};
+
+/**
+ * Exchange stSOL to SOL
+ * @param snapshot Snapshot of the Solido stats
+ * @param amount stSOL to exchange
+ */
+export const exchangeStSol = (snapshot: Snapshot, amount: StLamports) => {
+  const exchangeRate = getExchangeRate(snapshot);
+
+  return new Lamports(amount.stLamports.toNumber() * exchangeRate);
```

Comment:
We can be more precise here by using

```
(amount.stLamports * snapshot.solido.exchange_rate.sol_balance).toNumber() / snapshot.solido.exchange_rate.st_sol_supply.toNumber();
```

This way the conversion to float doesn’t happen until after multiplying the two numbers, and we don’t divide by a quotient.